scripts/seek/runParallelJobs.py

```diff
@@ -0,0 +1,124 @@
+import os
+import time
+import queue
+import threading
+import subprocess
+
+
+def funcThread(work_queue, tid=0, returnCodeQueue=None):
+    '''Runs python function jobs received from work_queue'''
+    # TODO - switch to runinng the functions using the multiprocessing library
+    #  to avoid GIL global thread interpreter lock contention
+    print('start thread')
+    try:
+        while job := work_queue.get(block=False):
+            args = job.get('args', ())
+            kwargs = job.get('kwargs', {})
+            func = job.get('func', None)
+            if func is None:
+                raise ValueError('Function not defined')
+            print(f'Thread {tid}: func: {func.__name__}, args: {args}, kwargs: {kwargs}')
+            func(*args, **kwargs)
+    except queue.Empty:
+        pass
+    return
```

Comment:
Not sure about this return. But maybe later you would like to add a status code?

Reply:
Yes, that's a good point. I have a return queue for jobThread, maybe I should add one here also.